src/components/Api.js

```diff
@@ -0,0 +1,86 @@
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseURL = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkTheApiResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserData() {
+    return fetch(`${this._baseURL}/users/me`, {
+      headers: this._headers,
+    }).then((res) => this._checkTheApiResponse(res));
```

Comment:
**COULD BE IMPROVED**

You can make a universal method `_request` not to duplicate the `ok` checking, `headers` and `baseUrl`


```js
 _request(endpoint, options = {}) {
    const finalOptions = {
      // add the default headers
      headers: this._headers,
      // spread the other options if provided
      ...options,
    };
      // add the endpoint to the base url
    const url = `${this._baseUrl}${endpoint}`;
    return fetch(url, finalOptions).then(this._checkResponse);
  }

  // usage

  getUserInfo() {
    // pass only the endpoint here
    return this._request("/users/me");
  }

  getInitialCards() {
    // pass only the endpoint here
    return this._request("/cards");
  }

  // and so on



```